baker/algolia/utils/mdimViews.ts

```diff
@@ -0,0 +1,159 @@
+        }
+        const params = new URLSearchParams(view.dimensions)
+        const queryStr = `?${params.toString()}`
+        const variableId = view.indicators.y[0].id
+        const metadata = merge(
+            relevantVariableMetadata[variableId],
+            multiDim.config.metadata,
+            view.metadata
+        )
+        const title =
+            metadata.presentation?.titlePublic ||
+            chartConfig.title ||
+            metadata.display?.name ||
+            metadata.name ||
+            ""
+        const subtitle = metadata.descriptionShort || chartConfig.subtitle || ""
+        const availableEntities = metadata.dimensions.entities.values
+            .map((entity) => entity.name)
+            .filter(Boolean)
+        const views_7d = pageviews[`/grapher/${slug}`]?.views_7d ?? 0
+        const score = views_7d * 10 - title.length
+        return {
+            type: ChartRecordType.MultiDimView,
+            objectID: `mdim-view-${id}`,
+            chartId: -1,
+            slug,
+            queryParams: queryStr,
+            title,
+            subtitle,
+            variantName: chartConfig.variantName,
+            keyChartForTags: [],
+            tags,
+            availableEntities,
+            publishedAt: new Date().toISOString(),
+            updatedAt: new Date().toISOString(),
+            numDimensions: chartConfig.dimensions?.length ?? 0,
+            titleLength: title.length,
+            numRelatedArticles: 0,
+            views_7d,
+            score,
+        }
+    })
+}
+
+async function getMultiDimDataPagesWithInheritedTags(
+    trx: db.KnexReadonlyTransaction
+) {
+    const multiDims = await getAllPublishedMultiDimDataPages(trx)
+    const parentTagArrays = await db.getParentTagArraysByChildName(trx)
+
+    const result = []
+    for (const multiDim of multiDims) {
+        const tags = multiDim.config.topicTags ?? []
+        if (tags.length === 0) {
+            await logErrorAndMaybeCaptureInSentry(
+                new Error(`MultiDim "${multiDim.slug}" has no tags.`)
+            )
+        }
+
+        const fullTags = new Set<string>()
+        for (const tag of tags) {
+            fullTags.add(tag)
+            const parentTags = getUniqueNamesFromParentTagArrays(
+                parentTagArrays[tag]
+            )
+            for (const parentTag of parentTags) {
+                fullTags.add(parentTag)
+            }
+        }
+
+        result.push({ multiDim, tags: [...fullTags] })
+    }
+
+    return result
+}
+
+export async function getMdimViewRecords(trx: db.KnexReadonlyTransaction) {
+    console.log("Getting mdim view records")
+    const multiDimsWithTags = await getMultiDimDataPagesWithInheritedTags(trx)
+    const pageviews = await getAnalyticsPageviewsByUrlObj(trx)
+    const records = await Promise.all(
+        multiDimsWithTags.map(({ multiDim, tags }) =>
+            getRecords(trx, multiDim, tags, pageviews)
+        )
+    )
+    return records.flat()
```

Comment:
```suggestion
    const records = await Promise.all(
        multiDimsWithTags.flatMap(({ multiDim, tags }) =>
            getRecords(trx, multiDim, tags, pageviews)
        )
    )
    return records
```

Reply:
`flatMap` [doesn't work](https://stackoverflow.com/q/74959270/9846837) in this context.